Svc/ComRetry/ComRetry.cpp

```diff
@@ -0,0 +1,67 @@
+// ======================================================================
+// \title  ComRetry.cpp
+// \author valdaarhun
+// \brief  cpp file for ComRetry component implementation class
+// ======================================================================
+
+#include "Svc/ComRetry/ComRetry.hpp"
+#include "ComRetry.hpp"
+
+namespace Svc {
+
+// ----------------------------------------------------------------------
+// Component construction and destruction
+// ----------------------------------------------------------------------
+
+ComRetry ::ComRetry(const char* const compName)
+    : ComRetryComponentBase(compName),
+      m_num_retries(1),
+      m_retry_count(0),
+      m_bufferState(Fw::Buffer::OwnershipState::OWNED) {}
+
+ComRetry ::~ComRetry() {}
+
+void ComRetry::configure(U32 num_retries) {
+    this->m_num_retries = num_retries;
+}
+
+// ----------------------------------------------------------------------
+// Handler implementations for typed input ports
+// ----------------------------------------------------------------------
+
+void ComRetry ::comStatusIn_handler(FwIndexType portNum, Fw::Success& condition) {
+    if (condition == Fw::Success::SUCCESS) {
+        this->m_retry_count = 0;
+        this->dataReturnOut_out(0, this->m_buffer, this->m_context);
+        this->comStatusOut_out(0, condition);
+    }
+    // Delivery of last message failed
+    else if (this->m_retry_count < this->m_num_retries) {
+        FW_ASSERT(this->m_bufferState == Fw::Buffer::OwnershipState::OWNED);
+        this->m_retry_count++;
+        this->m_bufferState = Fw::Buffer::OwnershipState::NOT_OWNED;
+        this->dataOut_out(0, this->m_buffer, this->m_context);
```

Comment:
We cannot retry on a failed message. This violates the expected protocol.  Instead, we need to wait until the next "SUCCESS" after a fail before returning.